src/definition/populate_test.go

```diff
@@ -516,15 +524,37 @@ func TestIntegrationProtocol2PopulateFunc_PopulateOnlySpecifiedGroups(t *testing
 	)(groups, withGeneratorSpec)
 	assert.True(t, populated)
 	assert.Empty(t, errs)
-	assert.Contains(t, integration.Data, &expectedEntityData1)
-	assert.Contains(t, integration.Data, &expectedEntityData2)
-	assert.Contains(t, integration.Data, &expectedEntityData3)
-	assert.Len(t, integration.Data, 3)
+	assert.Len(t, dummyIntgr.Entities, 3)
+
+	compareIgnoreFields := cmpopts.IgnoreUnexported(integration.Entity{}, metric.Set{}, inventory.Inventory{})
+
+	// Search for missing entities using cmp
+	for _, ti := range []*integration.Entity{
+		expectedEntityData1, expectedEntityData2, expectedEntityData3,
+	} {
+		found := false
+		diff := ""
+		for _, presentEntity := range dummyIntgr.Entities {
+			if cmp.Equal(presentEntity, ti, compareIgnoreFields) {
+				found = true
+				break
+			}
+			// Store "closest match" for better debuggability
+			curDiff := cmp.Diff(presentEntity, ti, compareIgnoreFields)
+			if len(diff) == 0 || len(curDiff) < len(diff) {
```

Comment:
`len(diff) == 0` should never happen here I think.

Reply:
I do `diff := ""` at the very beginning and then do the comparison to find the shortest diff. But if I skip that check, the empty var would always be the shortest diff :P

I'll add some comments to explain this better.

Reply:
Ooh, maybe if we name the vars `currentDiff` and `finalDiff`, the difference would be more obvious?

Or alternatively, you could collect all diffs, then find the smallest one?

Reply:
Renamed the vars and the comment to see if it helps, wdyt?

Reply:
Looks better! I have another round of review.